labs/FineTuning/HuggingFaceExample/02_inference/Inference-TinyLlama-1.1B.ipynb

```diff
@@ -0,0 +1,1467 @@
+    "```\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6d7a3c0e-556e-4f0f-b5eb-e9259bebc0db",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "s3_orig_model_path=\"\"  # <- change this path to your S3 model path from the Finetune notebook"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0c797454",
+   "metadata": {},
+   "source": [
+    "The settings in the container_arguments below for sequence length, batch size, and number of cores must match the settings you will use in your hub environment variables later.  The version of the SDK and Optimum Neuron must match as well, but we assue that by using the same container for both compilation as well as hosting."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "464f1220-a939-4779-86b2-15976f76348c",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "s3_output_path s3://sagemaker-us-east-2-445365384523/neuron_events2025/trn1-tinyllama-2025-05-13-00-40-31-750/output/model/compiled_model/\n",
+      "s3_model_path s3://sagemaker-us-east-2-445365384523/neuron_events2025/trn1-tinyllama-2025-05-13-00-40-31-750/output/model/merged_model/\n"
+     ]
+    }
+   ],
+   "source": [
+    "\n",
+    "# Define the parameters\n",
+    "s3_output_path=f\"{s3_orig_model_path}compiled_model/\"\n",
+    "print(\"s3_output_path\",s3_output_path)\n",
+    "training_job_name = utils.name_from_base(\"TGICompilation\")\n",
+    "print(\"training_job_name\")\n",
+    "s3_model_path = f\"{s3_orig_model_path}merged_model/\"\n",
+    "print(\"s3_model_path\",s3_model_path)\n",
+    "\n",
+    "container_entrypoint = [\"optimum-cli\"]\n",
+    "container_arguments = [\"export\", \"neuron\", \"--model\", \"/opt/ml/input/data/modeldir/\", \"--task\", \"text-generation\", \"--sequence_length\", \"512\", \"--batch_size\", \"1\", \"--num_cores\", \"2\", \"/opt/ml/output/data/\"]\n",
+    "\n",
+    "input_data_config = {\n",
+    "    \"ChannelName\": \"modeldir\",\n",
+    "    \"DataSource\": {\n",
+    "        \"S3DataSource\": {\n",
+    "            \"S3DataType\": \"S3Prefix\",\n",
+    "            \"S3Uri\": s3_model_path\n",
+    "        }\n",
+    "    }\n",
+    "}\n",
+    "output_data_config = {\n",
+    "    \"S3OutputPath\": s3_output_path\n",
+    "}\n",
+    "resource_config = {\n",
+    "    \"VolumeSizeInGB\": 20,\n",
+    "    \"InstanceCount\": 1,\n",
+    "    \"InstanceType\": \"ml.trn1.2xlarge\"\n",
+    "}\n",
+    "stopping_condition = {\n",
+    "    \"MaxRuntimeInSeconds\": 1800\n",
+    "}\n",
+    "\n",
+    "# Create the SageMaker client\n",
+    "sagemaker = boto3.client('sagemaker', region_name=sess.boto_session.region_name)\n",
+    "\n",
+    "# Create the training job\n",
+    "response = sagemaker.create_training_job(\n",
```

Comment:
Why are we switching to boto3 here? In the first notebook we use the SageMaker Python SDK.